File: FilteredData/filtercategory.py
import os
from collections import defaultdict

def extract_type(line):
    start = line.find('type="')
    if start == -1:
        return ""
    start += len('type="')
    end = line.find('"', start)
    if end == -1:
        return ""
    return line[start:end]

def split_by_category(input_file):
    if not os.path.exists(input_file):
        print("Girdi dosyası bulunamadı!")
        return

    file_handles = defaultdict(lambda: None)

    with open(input_file, 'r', encoding='utf-8') as infile:
        for line in infile:
            type_name = extract_type(line)
            if not type_name:
                continue

            file_name = f"{type_name}.txt"
            if file_handles[type_name] is None:
                file_handles[type_name] = open(file_name, 'w', encoding='utf-8')
            file_handles[type_name].write(line)

    for f in file_handles.values():
        if f:
            f.close()

    print("Tüm kayıtlar kategorilere göre ayrıldı.")

# Kullanım
split_by_category("filtered_by_date.txt")

File: FilteredData/filtercategory.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <unordered_map>
#include <filesystem>

std::string extractType(const std::string& line) {
    size_t start = line.find("type=\"");
    if (start == std::string::npos) return "";

    start += 6; // "type=\"" sonrası
    size_t end = line.find("\"", start);
    if (end == std::string::npos) return "";

    return line.substr(start, end - start);
}

int main() {
    std::ifstream inputFile("filtered_by_date.txt");
    if (!inputFile.is_open()) {
        std::cerr << "Girdi dosyası açılamadı!" << std::endl;
        return 1;
    }

    std::unordered_map<std::string, std::ofstream> categoryFiles;
    std::string line;

    while (std::getline(inputFile, line)) {
        std::string type = extractType(line);
        if (type.empty()) continue;

        // Dosya yoksa oluştur
        if (categoryFiles.find(type) == categoryFiles.end()) {
            std::string fileName = type + ".txt";
            categoryFiles[type].open(fileName);
            if (!categoryFiles[type].is_open()) {
                std::cerr << "Dosya açılamadı: " << fileName << std::endl;
                continue;
            }
        }

        // Satırı yaz
        categoryFiles[type] << line << std::endl;
    }

    // Tüm dosyaları kapat
    for (auto& [_, file] : categoryFiles) {
        file.close();
    }

    inputFile.close();

    std::cout << "Tüm kayıtlar kategorilere göre ayrıldı." << std::endl;
    return 0;
}
